From 797ef1098eaaa50c9bf6a3ac86ebfa511db756df Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] canvas File Patches


diff --git a/net/minecraft/world/level/block/Blocks.java b/net/minecraft/world/level/block/Blocks.java
index 3ee7f5c578e4721fd95390d76105d6f8976747a1..642344432c3e2a31448b203d47578cc2ee315c06 100644
--- a/net/minecraft/world/level/block/Blocks.java
+++ b/net/minecraft/world/level/block/Blocks.java
@@ -5983,6 +5983,12 @@ public class Blocks {
     private static Block register(
         final ResourceKey<Block> id, final Function<BlockBehaviour.Properties, Block> factory, final BlockBehaviour.Properties properties
     ) {
+        // Canvas start - MC-223153
+        if (io.canvasmc.canvas.GlobalConfiguration.getInstance().vanillaFixes.mc223153 && id.equals(BlockItemIds.RAW_COPPER_BLOCK.block())) {
+            Block block = factory.apply(properties.sound(SoundType.COPPER).setId(id));
+            return Registry.register(BuiltInRegistries.BLOCK, id, block);
+        }
+        // Canvas end - MC-223153
         Block block = factory.apply(properties.setId(id));
         return Registry.register(BuiltInRegistries.BLOCK, id, block);
     }
