From 797ef1098eaaa50c9bf6a3ac86ebfa511db756df Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] canvas File Patches


diff --git a/net/minecraft/world/level/block/EndGatewayBlock.java b/net/minecraft/world/level/block/EndGatewayBlock.java
index 1971099f8b815bdab711ca859f0ad45efd613dc8..b7c4a80a038cbb4dba54c1e031f799c590cd0f7a 100644
--- a/net/minecraft/world/level/block/EndGatewayBlock.java
+++ b/net/minecraft/world/level/block/EndGatewayBlock.java
@@ -96,6 +96,7 @@ public class EndGatewayBlock extends BaseEntityBlock implements Portal {
     ) {
         if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(level, pos)).callEvent()) { return; } // Paper - Add EntityInsideBlockEvent
         if (entity.canUsePortal(false)
+            && (!io.canvasmc.canvas.GlobalConfiguration.getInstance().blacklistNonPlayerEntitiesFromEnteringGatewayPortals || entity instanceof net.minecraft.server.level.ServerPlayer) // Canvas - blacklist non player entities from portals configs
             && !level.isClientSide()
             && level.getBlockEntity(pos) instanceof TheEndGatewayBlockEntity endGatewayBlockEntity
             && !endGatewayBlockEntity.isCoolingDown()) {
