From 797ef1098eaaa50c9bf6a3ac86ebfa511db756df Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] canvas File Patches


diff --git a/net/minecraft/data/advancements/packs/VanillaNetherAdvancements.java b/net/minecraft/data/advancements/packs/VanillaNetherAdvancements.java
index 6277cfc03f2a5cda0a8e8936c0e4876c4d790d7c..7a68f8906551f741f0716297614357d0e80a55a7 100644
--- a/net/minecraft/data/advancements/packs/VanillaNetherAdvancements.java
+++ b/net/minecraft/data/advancements/packs/VanillaNetherAdvancements.java
@@ -152,7 +152,7 @@ public class VanillaNetherAdvancements implements AdvancementSubProvider {
             .addCriterion(
                 "killed_ghast",
                 KilledTrigger.TriggerInstance.playerKilledEntity(
-                    EntityPredicate.Builder.entity().of(entityTypes, EntityTypes.GHAST).located(LocationPredicate.Builder.inDimension(Level.OVERWORLD))
+                    EntityPredicate.Builder.entity().of(entityTypes, EntityTypes.GHAST).located(LocationPredicate.Builder.inDimension(io.canvasmc.canvas.GlobalConfiguration.fetchRespawnDimensionKey())) // Canvas - default respawn dimension config - this is where the nether portal ties to, so we should change this to match that
                 )
             )
             .save(output, "nether/uneasy_alliance");
@@ -445,7 +445,7 @@ public class VanillaNetherAdvancements implements AdvancementSubProvider {
                 "ride_entity_distance",
                 DistanceTrigger.TriggerInstance.rideEntityInLava(
                     EntityPredicate.Builder.entity()
-                        .located(LocationPredicate.Builder.inDimension(Level.OVERWORLD))
+                        .located(LocationPredicate.Builder.inDimension(io.canvasmc.canvas.GlobalConfiguration.fetchRespawnDimensionKey())) // Canvas - default respawn dimension config - this is where the nether portal ties to, so we should change this to match that
                         .vehicle(EntityPredicate.Builder.entity().of(entityTypes, EntityTypes.STRIDER)),
                     DistancePredicate.horizontal(MinMaxBounds.Doubles.atLeast(50.0))
                 )
