From 797ef1098eaaa50c9bf6a3ac86ebfa511db756df Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] canvas File Patches


diff --git a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
index a13d8da418722b76b258016335d1095074756bee..ee94cd5a0f6e06f6fa7511ec9413bea318b9d218 100644
--- a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
@@ -303,11 +303,48 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     }
     // Canvas end - optimize flush calls with region threading
 
-    public void send(final Packet<?> packet) {
+    public void send(Packet<?> packet) { // Canvas - mutable method param
+        // Canvas start - no chat reports
+        if (io.canvasmc.canvas.GlobalConfiguration.getInstance().chat.disableChatReporting) {
+            Object self = this;
+            boolean cancel = false;
+
+            if (self instanceof ServerGamePacketListenerImpl listener
+                && packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat) {
+                packet = new net.minecraft.network.protocol.game.ClientboundSystemChatPacket(chat.chatType().decorate(
+                    chat.unsignedContent() != null ? chat.unsignedContent()
+                        : Component.literal(chat.body().content())
+                ), false);
+
+                cancel = true;
+                listener.send(packet);
+            }
+
+            if (cancel) {
+                return;
+            }
+        }
+        // Canvas end - no chat reports
         this.send(packet, null);
     }
 
     public void send(final Packet<?> packet, final @Nullable ChannelFutureListener listener) {
+        // Canvas start - no chat reports
+        if (io.canvasmc.canvas.GlobalConfiguration.getInstance().chat.disableChatReporting) {
+            Object self = this;
+            boolean cancel = false;
+
+            if (self instanceof ServerGamePacketListenerImpl listenerImpl
+                && packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat && listener != null) {
+                cancel = true;
+                listenerImpl.send(chat);
+            }
+
+            if (cancel) {
+                return;
+            }
+        }
+        // Canvas end - no chat reports
         // CraftBukkit start
         if (packet == null) {
             return;
